### ISSUE TITLE
Using Npcap 0.9995 results in BSOD.

### ISSUE THREAD

**Issue by l1TTss** [nmap/npcap#206]
Using Npcap 0.9995 results in a computer blue screen.(BSOD)
[071620-10562-01.zip](https://github.com/nmap/npcap/files/4929060/071620-10562-01.zip)

**Comment by dmiller-nmap** [comment]
Thanks for this report. I'm analyzing the crash right now; it looks like there's some corruption in the capture data structures. What were you doing just prior to the crash? Did you have more than one capture handle open, for example more than one instance of Wireshark running?

**Comment by l1TTss** [comment]
Using the Nessus (version：8.10.1)for vulnerability scanning.



------------------ Original ------------------

**Comment by dmiller-nmap** [comment]
I think I figured it out, but I'm not at my debugger to verify. The
pNBCopies object is refcounted twice, but may be freed/returned three
times. Need to check the proper order of object pool returns in
NPF_FreeCapData. Hopefully this will make sense to me in the morning.

>

**Comment by l1TTss** [comment]
Thank you very much for your effort.But newer versions often introduce new problems easily, and let's make it better later.

**Comment by l1TTss** [comment]
> I think I figured it out, but I'm not at my debugger to verify. The pNBCopies object is refcounted twice, but may be freed/returned three times. Need to check the proper order of object pool returns in NPF_FreeCapData. Hopefully this will make sense to me in the morning.
> […](#)
Thank you very much for your effort.But newer versions often introduce new problems easily, and let's make it better later.

**Comment by dmiller-nmap** [comment]
Upon further investigation, it does not seem that refcounting is the problem. The minidump does not have one of the important pages of memory for me to track everything that is going on, but here is the code in question, from `NPF_Read()`:

```C

		PNET_BUFFER pNetBuf = pCapData->pNBCopy->pNetBuffer;
		// NetBuffers that we create for copy don't use offsets and
		// begin right at the beginning of FirstMdl
		PMDL pMdl = NET_BUFFER_FIRST_MDL(pNetBuf);
```

The crash is a read access to 0x0000000000000020, which corresponds to the offset of `MdlChain` within the a `NET_BUFFER`, so `pNetBuf == NULL`, and I'm trying to determine how that could happen.

**Comment by dmiller-nmap** [comment]
The hang is a memory leak (#213) that may or may not be related to this crash. The code responsible for these structures has been extensively rewritten to be more efficient and avoid similar crashes. We expect to be able to make a new release this week.

**Comment by fyodor** [comment]
We believe this has been fixed in Npcap version 0.9996
